test/Microsoft.ML.AutoML.Tests/DatasetUtil.cs

```diff
@@ -82,5 +84,124 @@ private static string DownloadIfNotExists(string baseGitPath, string dataFile)
 
             throw new Exception($"Failed to download test file {dataFile}.");
         }
+
+        public static string GetFlowersDataset()
+        {
+            const string datasetName = @"flowers";
+            string assetsRelativePath = @"assets";
+            string assetsPath = GetAbsolutePath(assetsRelativePath);
+            string imagesDownloadFolderPath = Path.Combine(assetsPath, "inputs",
+                "images");
+
+            //Download the image set and unzip
+            string finalImagesFolderName = DownloadImageSet(
+                imagesDownloadFolderPath);
+
+            string fullImagesetFolderPath = Path.Combine(
+                imagesDownloadFolderPath, finalImagesFolderName);
+
+            var images = LoadImagesFromDirectory(
+                folder: fullImagesetFolderPath, useFolderNameAsLabel: true);
+
+            using (StreamWriter file = new StreamWriter(datasetName))
+            {
+                file.WriteLine("Label,ImagePath");
+                foreach (var image in images)
+                    file.WriteLine(image.Label + "," + image.ImagePath);
+            }
+
+            return datasetName;
+        }
+
+        public static IEnumerable<ImageData> LoadImagesFromDirectory(string folder,
+            bool useFolderNameAsLabel)
+        {
+            var files = Directory.GetFiles(folder, "*",
+                searchOption: SearchOption.AllDirectories);
+            foreach (var file in files)
+            {
+                if (Path.GetExtension(file) != ".jpg")
+                    continue;
+
+                var label = Path.GetFileName(file);
+                if (useFolderNameAsLabel)
+                    label = Directory.GetParent(file).Name;
+                else
+                {
+                    for (int index = 0; index < label.Length; index++)
+                    {
+                        if (!char.IsLetter(label[index]))
+                        {
+                            label = label.Substring(0, index);
+                            break;
+                        }
+                    }
+                }
+
+                yield return new ImageData()
+                {
+                    ImagePath = file,
+                    Label = label
+                };
+
+            }
+        }
+
+        public static string DownloadImageSet(string imagesDownloadFolder)
+        {
+            string fileName = "flower_photos_tiny_set_for_unit_tests.zip";
+            string url = $"https://mlnetfilestorage.file.core.windows.net/imagesets" +
+                $"/flower_images/flower_photos_tiny_set_for_unit_tests.zip?st=2019" +
+                $"-08-29T00%3A07%3A21Z&se=2030-08-30T00%3A07%3A00Z&sp=rl&sv=2018" +
+                $"-03-28&sr=f&sig=N8HbLziTcT61kstprNLmn%2BDC0JoMrNwo6yRWb3hLLag%3D";
```

Comment:
The new FlowerPhotosTiny dataset likely should have an entry added to the [test data README.md](https://github.com/dotnet/machinelearning/blob/056c60479304a3b5dbdf129c9bc6e853322bb090/test/data/README.md). 

While we are not redistributing the dataset from the repo, we are still redistributing the dataset from our stores (as opposed to downloading from a 3rd party). We don't otherwise have a place to mention the dataset source and the modifications we made to the dataset.

Reply:
Thanks for fixing.